src/react/reconcilation.js

```diff
@@ -1417,8 +1413,8 @@ export class Reconciler {
             throw new ExpectedBailOut(`abstract mapped arrays with "this" argument are not yet supported`);
           }
           this._queueOptimizedClosure(func, evaluatedNode, componentType, context, branchState);
-          return;
         }
+        return;
```

Comment:
Why did this return move?

Reply:
We can return earlier. This should have always been here really.